src/main/java/life/qbic/infrastructure/serialization/event/EventDeserializerFactory.java

```diff
@@ -0,0 +1,100 @@
+package life.qbic.infrastructure.serialization.event;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.IOException;
+import java.time.Instant;
+import life.qbic.domain.sample.SampleCode;
+import life.qbic.domain.sample.SampleEvent;
+import life.qbic.domain.sample.events.DataMadeAvailable;
+import life.qbic.domain.sample.events.FailedQualityControl;
+import life.qbic.domain.sample.events.LibraryPrepared;
+import life.qbic.domain.sample.events.MetadataRegistered;
+import life.qbic.domain.sample.events.PassedQualityControl;
+import life.qbic.domain.sample.events.SampleReceived;
+import life.qbic.exception.UnRecoverableException;
+
+/**
+ * <b>short description</b>
+ *
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
+public class EventDeserializerFactory {
+
+  private static final ObjectMapper objectMapper = configureObjectMapper();
+
+  public static <T extends SampleEvent> EventDeserializer<? extends SampleEvent> eventDeserializer(
+      Class<T> clazz) {
+    // decision on deserializer can be made based on clazz
+    // currently all events contain the same information so a generalization can be made.
```

Comment:
Yes, in case some property is added to an event (with a default value), the minor version would be increased. Thus the deserializer knows that there might be more fields that need to be populated. Currently, I did not implement that but the implementation should follow as soon as an individual event is changed.
You could for example have different (de)serializer for different versions. (e.g. one for all events of versions lower than 1.5)

Reply:
I see, I would prefer to then either implement something that showcases how the versioning is meant to be used or throw it out. I mean something like a version check. Otherwise, I don't see the point of having it now since it can be introduced when it is really needed.

Reply:
We need it now for use later. Otherwise, it will not be recorded in persistence.

Reply:
When we want to replay the events and they do not have a version, we will have problems.